Implemented Feature to download all Images by Public User on Insta. Fixed #16

devAyushDubey opened the pull request:
Hey @Mr-Sunglasses , I tried a few things, and it's working perfectly on my test bot.
Used threading to multitask fetching of images from Instagram.
fetch_posts function in [insta_all_pictures.py ](https://github.com/devAyushDubey/pastebin-telegram/blob/master/pastebin-bot/insta_all_pictures.py) uses multiple threads to download pictures from Instagram.
Invalid user name exception is also handled and a message is sent in response in the [bot.py](https://github.com/devAyushDubey/pastebin-telegram/blob/master/pastebin-bot/bot.py).

Mr-Sunglasses commented:
@devAyushDubey Do let me know if it is working without login using instagram.

Mr-Sunglasses commented on pastebin-bot/insta_all_pictures.py:
```diff
@@ -0,0 +1,27 @@
+def fetch_posts(user):
+
+    try:
+        prof = Profile.from_username(profile_instance.context, user)
```
As we are already using a meta, account for handling the username and password of insta to efficient use of Instaloader module so try to use `profile_instance.login(user=Username,passwd=Passoword)`

devAyushDubey commented:
> @devAyushDubey Do let me know if it is working without login using instagram.

Yes @Mr-Sunglasses this does not require any login.

Mr-Sunglasses commented:
@devAyushDubey I'm trying it to run locally and it is not working on my system, can you please share any demo that it is working in your system.

devAyushDubey commented:
> @devAyushDubey I'm trying it to run locally and it is not working on my system, can you please share any demo that it is working in your system.

You can test it on https://t.me/AyushOS_bot between 11 to 12:30, that's when I will get access to my laptop. I will run it locally so that you can test it. If you want API KEY to this bot for testing do let me know I can share that with you on anywhere private (Whatsapp /Discord).

Are you getting some sort of error btw?

Mr-Sunglasses commented:
> > @devAyushDubey I'm trying it to run locally and it is not working on my system, can you please share any demo that it is working in your system.
> 
> You can test it on https://t.me/AyushOS_bot between 11 to 12:30, that's when I will get access to my laptop. I will run it locally so that you can test it. If you want API KEY to this bot for testing do let me know I can share that with you on anywhere private (Whatsapp /Discord).
> 
> Are you getting some sort of error btw?

So the problem is The bot runs on the virtual machine in the server and when it request to Instagram and scrape the data, Instagram blocks it due to Proxy and IP address issues, that is why I'm asking you to use ID and password for scrapping. And also my computer exhausted all the requests limits 😅of Instaloader so it is not working on my computer.

Mr-Sunglasses commented:
![image](https://user-images.githubusercontent.com/81439109/190955530-4475daa8-577e-44a3-9e3c-30c277503eb4.png)

I think this is programmed differently 😂 😂

devAyushDubey commented:
> ![image](https://user-images.githubusercontent.com/81439109/190955530-4475daa8-577e-44a3-9e3c-30c277503eb4.png) @devAyushDubey I think this is programmed differently 😂 😂

I think we should ignore that @Mr-Sunglasses 😂. Just hostel things.

devAyushDubey commented:
I have added login in the last commit. @Mr-Sunglasses , do let me know if it works. And sorry for that greeting message, was not meant for you 😂, I hope it's chill.

Mr-Sunglasses commented:
![image](https://user-images.githubusercontent.com/81439109/191012439-345de8c4-72af-447e-9778-0c3087f1c944.png)

It is just not working on my device 🙃.

devAyushDubey commented:
Yeah @Mr-Sunglasses  I was doing some changes that time. You should checkout the [test bot](https://t.me/AyushOS_bot) now. I hosted it on a server and made some changes. Of course added Login, and Rate Limit, I am limiting it to usernames with <= 200 posts.
Instaloader and Instagram API behaves weird sometimes, there were many issues for the same on Instaloader's repo too. While the bot would work fine, there could be 400 and 401 sometimes randomly. I have added different delays before every request to minimize this. Try it on the test bot and do let me know so that I can make a commit.

And yeah I changed that greeting message 🤦‍♀️ to Hi User! 😂

devAyushDubey commented:
![WhatsApp Image 2022-09-19 at 10 53 35 PM](https://user-images.githubusercontent.com/33064931/191084723-3cb838af-57aa-433e-a4c0-3802792d7eca.jpeg)

Mr-Sunglasses reviewed (approved):
Looking Good we can now merge it.

AnjumanHasan reviewed (approved):
LGTM

Mr-Sunglasses commented:
Thanks @devAyushDubey , Will Looking forward to Improve this feature in the next version.

devAyushDubey commented:
![Screenshot (950)](https://user-images.githubusercontent.com/33064931/191258334-6aa2a16a-90a6-4717-a733-c8ea59c51e3d.png)
@Mr-Sunglasses Just got my account blocked (not main one) 😂😂. I had that last bit of code that added that rate limit and delays, it also optimized the login but I don't think I would be able to test it now 😂, so let that be as it is for now.

Was nice contributing to pastebot 😉.
